File: ruby/council/if_else.rb
'case'

class SithInDisguisePresenter
  def initialize(jedi)
    @jedi = jedi
  end

  def alterior_motives
  end
end

class HighCouncilController < ApplicationController
  def index
    @jedi = Jedi.all

    @jedi.map do |jedi|
      prepare_for_meeting(jedi)
    end
  end

  private

  def prepare_for_meeting(jedi)
    if jedi.is_palpatine?
      SithInDisguise.new(jedi)
    else
      jedi
    end
  end
end

File: csharp/council/if_else.cs
"type_checking.cs"

class SithInDisguiseDecorator
{
  public SithInDisguiseDecorator(Jedi jedi)
  {
    JediToPresent = jedi;
  }

  public Jedi JediToPresent { get; set; }

  public string AlteriorMotives() {}
  public int CountOfPeopleBeingManipulated() {}
}

class HighCouncilController : Controller
{
  public ActionResult Index()
  {
    List<Jedi> list = db.Jedi.ToList();

    list.Select(presented => PrepareForMeeting(jedi));

    return View();
  }

  private PrepareForMeeting(Jedi jedi)
  {
    if(jedi.IsPalpatine())
    {
      return new SithInDisguiseDecorator(jedi);
    } else {
      return jedi;
    }
  }
}
